sw/device/silicon_creator/manuf/lib/individualize.c

```diff
@@ -33,64 +33,47 @@ typedef struct hw_cfg0_settings {
-   */
-  multi_bit_bool_t en_entropy_src_fw_read;
-  /**
-   * This input efuse is used to enable access to the firmware override FIFO and
-   * other related functions.
-   */
-  multi_bit_bool_t en_entropy_src_fw_over;
-} hw_cfg0_settings_t;
+} hw_cfg1_settings_t;
 
 // Changing any of the following values may result in unexpected device
 // behavior.
 //
 // - en_sram_ifetch: required to enable SRAM code execution during
 //   manufacturing.
 // - en_csrng_sw_app_read: required to be able to extract output from CSRNG.
-// - en_entropy_src_fw_read and en_entropy_src_fw_over: Required to implement
-//   entropy_src conditioner KAT.
-const hw_cfg0_settings_t kHwCfg0Settings = {
+const hw_cfg1_settings_t kHwCfg1Settings = {
     .en_sram_ifetch = kMultiBitBool8True,
-    .en_csrng_sw_app_read = kMultiBitBool8True,
-    .en_entropy_src_fw_read = kMultiBitBool8True,
-    .en_entropy_src_fw_over = kMultiBitBool8True,
-};
+    .en_csrng_sw_app_read = kMultiBitBool8True};
 
 /**
- * Configures digital logic settings in the HW_CFG0 partition.
+ * Configures digital logic settings in the HW_CFG1 partition.
  *
  * @param otp OTP controller instance.
  * @return The result of the operation.
  */
 OT_WARN_UNUSED_RESULT
-static status_t hw_cfg0_enable_knobs_set(const dif_otp_ctrl_t *otp_ctrl) {
+static status_t hw_cfg1_enable_knobs_set(const dif_otp_ctrl_t *otp_ctrl) {
   uint32_t val =
-      bitfield_field32_write(0, kSramFetch, kHwCfg0Settings.en_sram_ifetch);
+      bitfield_field32_write(0, kSramFetch, kHwCfg1Settings.en_sram_ifetch);
   val = bitfield_field32_write(val, kCsrngAppRead,
-                               kHwCfg0Settings.en_csrng_sw_app_read);
-  val = bitfield_field32_write(val, kEntropySrcFwRd,
-                               kHwCfg0Settings.en_entropy_src_fw_read);
-  val = bitfield_field32_write(val, kEntropySrcFwOvr,
-                               kHwCfg0Settings.en_entropy_src_fw_over);
+                               kHwCfg1Settings.en_csrng_sw_app_read);
 
-  TRY(otp_ctrl_testutils_dai_write32(otp_ctrl, kDifOtpCtrlPartitionHwCfg0,
+  TRY(otp_ctrl_testutils_dai_write32(otp_ctrl, kDifOtpCtrlPartitionHwCfg1,
                                      kHwCfgEnSramIfetchOffset, &val,
                                      /*len=*/1));
   return OK_STATUS();
 }
 
-status_t manuf_individualize_device_hw_cfg0(
+status_t manuf_individualize_device_hw_cfg(
     dif_flash_ctrl_state_t *flash_state, const dif_otp_ctrl_t *otp_ctrl,
     dif_flash_ctrl_region_properties_t flash_info_page_0_permissions,
     uint32_t *device_id) {
-  bool is_locked;
+  bool is_locked0, is_locked1;
   TRY(dif_otp_ctrl_is_digest_computed(otp_ctrl, kDifOtpCtrlPartitionHwCfg0,
-                                      &is_locked));
-  if (is_locked) {
+                                      &is_locked0));
+  TRY(dif_otp_ctrl_is_digest_computed(otp_ctrl, kDifOtpCtrlPartitionHwCfg1,
+                                      &is_locked1));
+  if (is_locked0 && is_locked1) {
```

Comment:
perhaps we should just skip provisioning HwCfg0 if HwCfg0 is locked, and HwCfg1 if HwCfg1 is locked, rather than only skipping the provisioning if both is locked? this would then handle the case where provisioning one partition succeeds but the other fails (we aim for this code to be reentrant).

Reply:
Ah duh, makes total sense :).
I changed it, PTAL.